src/components/8Ball/8Ball.js

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { fetchMessage } from '../../helper-fns/apiCalls';
+import eightBallFilled from '../../images/eightBallFilled.png';
+import eightBallBLUE from '../../images/eightBallBLUE.png';
+import './8Ball.css';
+
+export const EightBall = ({ cityName }) => {
+  const [message, setMessage] = useState('');
+  const [error, setError] = useState(false);
+  const [shaking, setShaking] = useState(false);
+  const [count, setCount] = useState(0);
+
+  const handleClick = async e => {
+    e.preventDefault();
+    setMessage('');
+    setError('');
+    setCount(count + 1);
+    await shakeBall(2000);
+    try {
+      let answer = await fetchMessage('Should I move?');
+      setMessage(answer);
+      setShaking(false);
+    } catch (err) {
+      setError(err.message);
+    }
+  };
+
+  const shakeBall = ms => {
+    setShaking(true);
+    return new Promise(resolve => setTimeout(resolve, ms));
+  };
+
+  return (
+    <section className='magic-8-ball'>
```

Comment:
This is a great use of semantic tags and error handling logic to render the various 8ball options!